tests/snapshots.rs

```diff
@@ -619,11 +629,33 @@ fn convert_wgsl() {
         let file = fs::read_to_string(format!("{root}/{BASE_DIR_IN}/{name}.wgsl"))
             .expect("Couldn't find wgsl file");
         match naga::front::wgsl::parse_str(&file) {
-            Ok(module) => check_targets(&module, name, targets, None),
+            Ok(module) => check_targets(&module, name, targets, &[], None),
             Err(e) => panic!("{}", e.emit_to_string(&file)),
         }
     }
-
+    /* These tests needs some capabilities to be present */
+    {
+        let inputs = [(
+            "dualsource",
+            Targets::SPIRV | Targets::METAL | Targets::GLSL | Targets::HLSL | Targets::WGSL,
+        )];
+        for &(name, targets) in inputs.iter() {
+            println!("Processing '{name}'");
+            // WGSL shaders lives in root dir as a privileged.
+            let file = fs::read_to_string(format!("{root}/{BASE_DIR_IN}/{name}.wgsl"))
+                .expect("Couldn't find wgsl file");
+            match naga::front::wgsl::parse_str(&file) {
+                Ok(module) => check_targets(
+                    &module,
+                    name,
+                    targets,
+                    &[naga::valid::Capabilities::DUAL_SOURCE_BLENDING],
+                    None,
+                ),
+                Err(e) => panic!("{}", e.emit_to_string(&file)),
+            }
+        }
+    }
```

Comment:
Let's specify `god_mode` in the `.param.ron` file of this snapshot to keep things consistent.

I opened https://github.com/gfx-rs/naga/issues/2442 to track passing finer grained `Capabilities`.